### ISSUE TITLE
[Accessibility]: Featured extensions do not respect filters in dropdown under All Category

### ISSUE THREAD

**Issue by abeatrix** [sourcegraph/sourcegraph#34799]
### Audit type

General/Other

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/34052

### Problem description

An issue was identified while auditing https://github.com/sourcegraph/sourcegraph/issues/34052

> Use the "Show all" dropdown in the sidebar to select whether to display enabled, disabled, or all extensions, and whether to hide experimental extensions.

When the `All` Category is selected:
Clicking on `Show enabled extensions` from the  dropdown in the sidebar would still show disabled extension from the `Featured` tab:
![image](https://user-images.githubusercontent.com/68532117/166318285-e635f99b-85bb-4007-8aa5-ab2c64604fef.png)

Clicking on  `Show disabled extensions` from the  dropdown in the sidebar would still show enabled extension from the `Featured` tab:
![image](https://user-images.githubusercontent.com/68532117/166318350-17277816-5b7f-41b2-a5c4-9724c94a95e9.png)

This does not align with what the dropdown button suggests and create confusion

### Expected behavior

Only enabled extensions would show up when `Show enabled extensions` is clicked and vice versa. 

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [ ] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.

**Comment by sourcegraph-bot-2** [comment]
Heads up @muratsu @jjinnii @ryankscott - the "team/integrations" label was applied to this issue.
